flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/SerializableAccessExecutionGraph.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.executiongraph;
+
+import java.io.Serializable;
+
+/**
+ * {@code SerializableAccessExecutionGraph} is a serializable version of {@link
+ * AccessExecutionGraph}.
+ */
+public interface SerializableAccessExecutionGraph extends AccessExecutionGraph, Serializable {}
```

Comment:
This would now make 4 interfaces that are exclusively for 2 classes?

Interfaces: { `JobStatusProvider`, `AccessExecutionGraph`, `SerializableAccessExecutionGraph`, `ExecutionGraph` }
Classes: { `DefaultExecutionGraph`, `ArchivedExecutionGraph` }

Perhaps it is time to restrategize about refactoring this. `AccessExecutionGraph` is both a sort of "readonly" ExecutionGraph interface and also a placeholder for the `ArchivedExecutionGraph` but it can't really be both since an `ArchivedExecutionGraph` is strictly not an `ExecutionGraph`? I am not sure what to suggest and I can certainly see the logic in `AccessExecutionGraph` as a "readonly" `ExecutionGraph` interface.

Reply:
Thanks for sharing your perspective, @bytesandwich . You're right - the `ExecutionGraph`- and `Execution`-related classes/interfaces are up for discussion already for a while as they make testing individual code harder right now.

My proposal of introducing `SerializableAccessExecutionGraph` was solely based on considerations around [FLINK-21439](https://issues.apache.org/jira/browse/FLINK-21439) as stated in [my comment above](https://github.com/apache/flink/pull/15049#discussion_r598459965). But I am happy to revert the switch from `ArchivedExecutionGraph` to `AccessExecutionGraph` as the change might not be necessary depending on how we tackle `FLINK-21439` in the end.